// node-zoho/build/products/crm/Accounts.js
// Generated by CoffeeScript 1.7.1
var Accounts, CrmModule, Request, xml2js, _,
  __hasProp = {}.hasOwnProperty,
  __extends = function(child, parent) { for (var key in parent) { if (__hasProp.call(parent, key)) child[key] = parent[key]; } function ctor() { this.constructor = child; } ctor.prototype = parent.prototype; child.prototype = new ctor(); child.__super__ = parent.prototype; return child; };

_ = require('underscore');

xml2js = require("xml2js");

CrmModule = require('./crm-module');

Request = require('../../request');

Accounts = (function(_super) {
  __extends(Accounts, _super);

  function Accounts() {
    return Accounts.__super__.constructor.apply(this, arguments);
  }

  Accounts.prototype.name = 'Accounts';

  Accounts.prototype.getMyRecords = function() {
    throw new Error('Not Implemented');
  };

  Accounts.prototype.getSearchRecordsByPDC = function() {
    throw new Error('Not Implemented');
  };

  Accounts.prototype.deleteRecords = function() {
    throw new Error('Not Implemented');
  };

  Accounts.prototype.convertLead = function() {
    throw new Error('Not Implemented');
  };

  Accounts.prototype.getRelatedRecords = function() {
    throw new Error('Not Implemented');
  };

  Accounts.prototype.updateRelatedRecords = function() {
    throw new Error('Not Implemented');
  };

  Accounts.prototype.getUsers = function() {
    throw new Error('Not Implemented');
  };

  Accounts.prototype.uploadFile = function() {
    throw new Error('Not Implemented');
  };

  Accounts.prototype.downloadFile = function() {
    throw new Error('Not Implemented');
  };

  Accounts.prototype.deleteFile = function() {
    throw new Error('Not Implemented');
  };

  Accounts.prototype.uploadPhoto = function() {
    throw new Error('Not Implemented');
  };

  Accounts.prototype.downloadPhoto = function() {
    throw new Error('Not Implemented');
  };

  Accounts.prototype.deletePhoto = function() {
    throw new Error('Not Implemented');
  };

  return Accounts;

})(CrmModule);

module.exports = Accounts;
